debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java

```diff
@@ -0,0 +1,202 @@
+                    Metronome metronome = Metronome.sleeper(Duration.ofSeconds(1L), clock);
+                    metronome.pause();
+                }
+
+                LogMinerHelper.startOnlineMining(connection, lastProcessedScn, currentScn);
+
+                fetchChangesFromMiningView.setLong(1, lastProcessedScn);
+                fetchChangesFromMiningView.setLong(2, lastProcessedScn);
+
+                debugInfo(connection, "before");
+                res = fetchChangesFromMiningView.executeQuery();
+                debugInfo(connection, "after");
+
+                dispatchChanges(res);
+                offsetContext.setScn(currentScn);
+
+                res.close();
+            }
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        } finally {
+            // 3. disconnect
+            try (Connection connection = jdbcConnection.connection()) {
+                LogMinerHelper.endMining(connection);
+            } catch (SQLException e) {
+                LOGGER.error("Cannot close Log Miner session gracefully: " + e.getMessage());
+            }
+            if (res != null) {
+                try {
+                    res.close();
+                } catch (SQLException e) {
+                    LOGGER.error("Cannot close result set due to the :" + e.getMessage());
+                }
+            }
+            try {
+                jdbcConnection.close();
+            } catch (SQLException e) {
+                LOGGER.error("Cannot close JDBC connection: " + e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void commitOffset(Map<String, ?> offset) {
+        // todo nothing here?
+    }
+
+    private void dispatchChanges(ResultSet res) throws SQLException {
+
+        LOGGER.debug("dispatching changes..");
+        List<LogMinerRowLcr> changes = logMinerProcessor.process(res);
+
+        for (LogMinerRowLcr logMinerRowLcr : changes) {
+
+            long actualScn = logMinerRowLcr.getActualScn();
+            offsetContext.setScn(actualScn);
+            //offsetContext.setLcrPosition(new LcrPosition());
+            offsetContext.setTransactionId(logMinerRowLcr.getTransactionId());
+            offsetContext.setSourceTime(logMinerRowLcr.getSourceTime().toInstant());
+            offsetContext.setTableId(getTableId(logMinerRowLcr));
+            try {
+                LOGGER.debug("Processing DML event {} scn {}", logMinerRowLcr, actualScn);
+                TableId tableId = getTableId(logMinerRowLcr);
+                dispatcher.dispatchDataChangeEvent(
+                        tableId,
+                        new ChangeRecordEmitter(offsetContext, logMinerRowLcr, schema.tableFor(tableId), clock)
+                );
+            } catch (InterruptedException e) {
+                LOGGER.error("Following SCN: " + logMinerRowLcr.getActualScn() + " cannot be dispatched due to the : " + e);
```

Comment:
`InterruptedException` should rather be let passed up to the caller which handles it appropriately.